src/main/java/wolox/training/models/Book.java

```diff
@@ -126,6 +133,7 @@ public String getYear() {
     }
 
     public void setYear(String year) {
+        Preconditions.checkNotNull(year);
```

Comment:
After checking if a field is null, check if the field is empty, and then check if it is numeric.

An error message can be added to each check :

```
Preconditions.checkNotNull(year, "The year field cannot be null.");
Preconditions.checkArgument(!year.isEmpty(), "The year field cannot be empty.");
Preconditions.checkArgument(StringUtils.isNumeric(year), "error message");
this.year = year;
```

I recommend using a class to store error messages as constants.

Reply:
I put message in constant and added validation numeric